src/libsync/propagatedownload.cpp

```diff
@@ -344,6 +344,42 @@ void PropagateDownloadFile::start()
         }
     }
 
+    // If we have a conflict where size and mtime are identical,
+    // compare the remote checksum to the local one.
+    // Maybe it's not a real conflict and no download is necessary!
+    QByteArray checksumType = parseChecksumHeaderType(_item->_checksumHeader);
+    if (_item->_instruction == CSYNC_INSTRUCTION_CONFLICT
+        && _item->_size == _item->log._other_size
+        && _item->_modtime == _item->log._other_modtime
+        && collisionSafeHashes().contains(checksumType)) {
```

Comment:
How about doing it even if it's not a collisionsafe hash, it would still be better than before where we were not comparing anything.

Reply:
Adler32 is not good enough

Reply:
You're missing the point, it goes the other way.
We were considering the file to be equal if they have the same size and mtime.
When there is a checksum, we now add a security check by comparing the checksum.